### PR TITLE
[TIMOB-19580] Customize Activity Transitions

### PR DESCRIPTION
**Pull request by sriks**
JIRA https://jira.appcelerator.org/browse/TIMOB-19580?filter=-1

### CODE REVIEW

**Comment by ashcoding** [comment]
Just glancing through the code. There seems to be some slight formatting issues. Will comment on those parts of the code.

File: android/titanium/src/java/org/appcelerator/titanium/view/TiUIView.java [hunk 1]
```diff
@@ -91,6 +93,19 @@
 	public static final int SOFT_KEYBOARD_DEFAULT_ON_FOCUS = 0;
 	public static final int SOFT_KEYBOARD_HIDE_ON_FOCUS = 1;
 	public static final int SOFT_KEYBOARD_SHOW_ON_FOCUS = 2;
+	
+    public static final int TRANSITION_NONE                     =   0;
```
Comment (ashcoding) [inline android/titanium/src/java/org/appcelerator/titanium/view/TiUIView.java]: The variables are not formatted with the variables above.

File: android/titanium/src/java/org/appcelerator/titanium/proxy/TiWindowProxy.java [hunk 2]
```diff
@@ -481,4 +491,39 @@ public KrollProxy getParentForBubbling()
...
+	    if (LOLLIPOP_OR_GREATER) {
+	        TiUIView v = view.peekView();
+	        if (v != null) {
+	            Pair< View,String > p = new Pair<View, String>(v.getNativeView(), transitionName);
+	            sharedElementPairs.add(p);
+	        }
+	    }
+	}
+	
+	@Kroll.method
```
Comment (ashcoding) [inline android/titanium/src/java/org/appcelerator/titanium/proxy/TiWindowProxy.java]: Annotation not formatted correctly.

File: android/titanium/src/java/org/appcelerator/titanium/TiC.java [hunk 3]
```diff
@@ -838,6 +846,11 @@
 	 */
 	public static final String PROPERTY_BACKGROUND_SELECTED_IMAGE = "backgroundSelectedImage";
 	
+    /**
+     * @module.api
+     */
+    public static final String PROPERTY_TRANSITION_NAME = "transitionName";
```
Comment (ashcoding) [inline android/titanium/src/java/org/appcelerator/titanium/TiC.java]: Variable and comments not formatted correctly

File: android/modules/ui/src/java/ti/modules/titanium/ui/android/AndroidModule.java [hunk 4]
```diff
@@ -81,10 +81,19 @@
 	@Kroll.constant public static final int WEBVIEW_LOAD_CACHE_ONLY = WebSettings.LOAD_CACHE_ONLY;
 	@Kroll.constant public static final int WEBVIEW_LOAD_CACHE_ELSE_NETWORK = WebSettings.LOAD_CACHE_ELSE_NETWORK;
 
-
-
-
-
+    @Kroll.constant public static final int TRANSITION_NONE                     =   TiUIView.TRANSITION_NONE;
```
Comment (ashcoding) [inline android/modules/ui/src/java/ti/modules/titanium/ui/android/AndroidModule.java]: Variable and comments not formatted correctly

File: android/modules/ui/src/java/ti/modules/titanium/ui/WindowProxy.java [hunk 5]
```diff
@@ -455,7 +471,116 @@ private void setWindowWidthHeight(Object width, Object height)
...
+	
+	/**
+	 * Helper method to apply activity transitions.
+	 * @param win The window holding the activity.
+	 * @param props The property dictionary. 
+	 */
+	private void applyActivityTransitions(Window win, KrollDict props) {
+		if (LOLLIPOP_OR_GREATER) {
+			// Return and reenter transitions defaults to enter and exit transitions respectively only if they are not set.
+		    // And setting a null transition makes the view unaccounted from transition.
```
Comment (ashcoding) [inline android/modules/ui/src/java/ti/modules/titanium/ui/WindowProxy.java]: Comments not formatted correctly

**Comment by sriks** [comment]
Please review the updated commit.
Though not seeing formatting difference in eclipse, explicitly applied indentation. Looks like git diff is sensitive to tabs!

**Comment by ashcoding** [comment]
Code reviewed. Functionally tested. **Accepted**

**Pull request merged**